Add object_list to BaseListView

# Bug report

<!--
Hi, thanks for submitting a bug. We appreciate that.

But, we will need some information about what's wrong to help you.
-->

## What's wrong

When accessing `self.object_list` in a class inheriting from `ListView` implementation, mypy reports that the attribute does not exist:

```
$ mypy testcase.py
testcase.py:18: error: "MyListView" has no attribute "object_list"  [attr-defined]
            context["objects"] = self.object_list
                                 ^
Found 1 error in 1 file (checked 1 source file)
```

Corresponding code:
```py
from __future__ import annotations

from typing import Any, Iterable

from django.db.models import Model
from django.views.generic import ListView


class MyModel(Model):
    pass


class MyListView(ListView):
    model = MyModel

    def get_context_data(self, *, object_list: Iterable[MyModel] | None = None, **kwargs: Any) -> dict[str, Any]:
        context = super().get_context_data(object_list=object_list, **kwargs)
        context["objects"] = self.object_list
        return context
```

## How is that should be

While `object_list` isn't defined at construction, `BaseListView.get()` sets it when handling a request, so in practice it is defined whenever a list view is doing its work. Therefore I think it makes sense to add it to the stub of `BaseListView`.

## System information

- OS: Ubuntu Linux
- `python` version: 3.9.7
- `django` version: 3.2.10
- `mypy` version: 0.920
- `django-stubs` version: 1.9.0
- `django-stubs-ext` version: 0.3.1

--- a/django-stubs/views/generic/list.pyi
+++ b/django-stubs/views/generic/list.pyi
@@ -30,6 +30,7 @@ class MultipleObjectMixin(Generic[T], ContextMixin):
     def get_context_object_name(self, object_list: QuerySet) -> Optional[str]: ...
 
 class BaseListView(MultipleObjectMixin[T], View):
+    object_list: Sequence[T]
     def get(self, request: HttpRequest, *args: Any, **kwargs: Any) -> HttpResponse: ...
 
 class MultipleObjectTemplateResponseMixin(TemplateResponseMixin):

